app/scripts/controllers/metametrics.ts

```diff
@@ -80,55 +109,129 @@ const buildUniqueMessageId = (args) => {
   return generateRandomId();
 };
 
-const exceptionsToFilter = {
+const exceptionsToFilter: Record<string, boolean> = {
   [`You must pass either an "anonymousId" or a "userId".`]: true,
 };
 
-/**
- * @typedef {import('../../../shared/constants/metametrics').MetaMetricsContext} MetaMetricsContext
- * @typedef {import('../../../shared/constants/metametrics').MetaMetricsEventPayload} MetaMetricsEventPayload
- * @typedef {import('../../../shared/constants/metametrics').MetaMetricsEventOptions} MetaMetricsEventOptions
- * @typedef {import('../../../shared/constants/metametrics').SegmentEventPayload} SegmentEventPayload
- * @typedef {import('../../../shared/constants/metametrics').SegmentInterface} SegmentInterface
- * @typedef {import('../../../shared/constants/metametrics').MetaMetricsPagePayload} MetaMetricsPagePayload
- * @typedef {import('../../../shared/constants/metametrics').MetaMetricsPageOptions} MetaMetricsPageOptions
- * @typedef {import('../../../shared/constants/metametrics').MetaMetricsEventFragment} MetaMetricsEventFragment
- * @typedef {import('../../../shared/constants/metametrics').MetaMetricsTraits} MetaMetricsTraits
- */
+type SegmentEventType = 'identify' | 'track' | 'page';
```

Comment:
Should we document that this should correspond to one of the methods in the Analytics class?

``` suggestion
/**
 * The type of a Segment event to create.
 *
 * Must correspond to the name of a method in {@link Analytics}.
 */
type SegmentEventType = 'identify' | 'track' | 'page';